clash-lib/src/Clash/Core/Subst.hs

```diff
@@ -765,13 +771,18 @@ freshenTm is0 = go (mkSubst is0) where
        (is1, e') -> (is1, Tick (goTick subst0 tick) e')
     tm -> (substInScope subst0, tm)
 
-  goBind subst0 xs =
+  goBind subst0 (NonRec i x) =
+    let (subst1, i') = substIdBndr subst0 i
+        (is2, x') = go subst1 x
+     in (subst1 { substInScope = is2}, NonRec i' x')
```

Comment:
Shouldn't this be?
```suggestion
    let (subst1, i') = substIdBndr subst0 i
        (is2, x') = go subst x
     in (subst1 { substInScope = extendInScopeSet is2 i'}, NonRec i' x')
```
Since `i` doesn't actually scope over `x`.